src/wallet/rpcwallet.cpp

```diff
@@ -106,7 +106,7 @@ std::shared_ptr<CWallet> GetWalletForJSONRPCRequest(const JSONRPCRequest& reques
         return wallets[0];
     }
 
-    if (!HasWallets()) {
+    if (wallets.empty()) {
```

Comment:
Why it is safe do not lock `cs_wallets` now?

Reply:
I presume for the same reason that it is safe to return wallets[0] without the lock. cs_wallets only protects vpwallets, not copies of the shared pointers.